src/modules/auth/token.service.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { Controller, Get, INestApplication, Injectable, InternalServerErrorException, MiddlewareConsumer, Module, RequestMethod, UnauthorizedException } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Request } from 'express';
+import { NextFunction } from 'express-serve-static-core';
+import request from 'supertest';
+import * as jwt from 'jsonwebtoken';
+import { TokenPayload } from './token.service';
+
+// Mock token service class
+@Injectable()
+class TokenService {
+
+    handleOriginCheck = async (req: Request, res: Response, next: NextFunction) => {
+        let token = null;
+        if (req.headers['authorization']) {
+            token = req.headers['authorization'].replace('Bearer ', '');
+        } else {
+            token = req?.cookies?.token
+        }
+
+        if (token) {
+            const decodedToken = jwt.decode(token) as TokenPayload;
+            const isBrowserRequestFromAuthenticatedOrigin = decodedToken.origin === req.headers['origin']
+            const isServerRequestOrGETFromSameDomain = req.headers['origin'] === undefined
+            if (isBrowserRequestFromAuthenticatedOrigin || isServerRequestOrGETFromSameDomain) {
+                next()
+            } else {
+                throw new InternalServerErrorException('Something went wrong')
+            }
+        } else {
+            throw new UnauthorizedException('Unauthorized')
+        }
+    }
+}
+
+// mock controller
+@Controller()
+class TestController {
+    @Get('test')
+    test() {
+        return "RETURN_VALUE";
+    }
+}
+
+// mock the middleware
+@Module({ providers: [TokenService], exports: [TokenService] })
+class TestModule {
+    constructor(private readonly tokenService: TokenService) { }
+
+    configure(consumer: MiddlewareConsumer) {
+        consumer
+            .apply((req, res, next) => this.tokenService.handleOriginCheck(req, res, next))
+            .exclude(
+                { path: '/login', method: RequestMethod.ALL },
+                { path: '/refresh_token', method: RequestMethod.ALL }
+            )
+            .forRoutes({ path: '/*', method: RequestMethod.ALL });
+    }
+};
+
+describe('TokenService', () => {
```

Comment:
Would it be possible to add a test that demonstrates that the token check fails if the origin header doesn't match the token? Or is that not possible because it's not possible to set the origin header in the test?

Reply:
+1

Reply:
sure its possible just by setting  `.set({ 'origin': 'anydomain.com' })` [here](https://github.com/energywebfoundation/iam-cache-server/blob/3e8a70a081fa209b3106761955cd40cc9e47a70e/src/modules/auth/token.service.spec.ts#L79), so if origin in headers will no match with origin in token it will throw 500 internal server error.

Reply:
updated by latest [commit](https://github.com/energywebfoundation/iam-cache-server/pull/145/commits/34318563453d02a5a399f48163b79890cb244c15)